packages/cf.js/src/app-factory.ts

```diff
@@ -1,5 +1,42 @@
-import { AppDefinition } from "./structs";
+import {
+  NodeMessageType,
+  NodeProposeInstallData
+} from "@counterfactual/node-provider";
+import { BigNumber, BigNumberish } from "ethers/utils";
+
+import { AppDefinition, BlockchainAsset } from "./app-types";
+import { Provider } from "./provider";
+import { Address, AppInstanceID } from "./simple-types";
+
+export interface InstallProposalRequest {
+  peerAddress: Address;
+  asset: BlockchainAsset;
+  myDeposit: BigNumberish;
+  peerDeposit: BigNumberish;
+  initialState: any;
+}
 
 export class AppFactory {
-  constructor(readonly appDefinition: AppDefinition) {}
+  constructor(
+    readonly provider: Provider,
+    readonly appDefinition: AppDefinition
+  ) {}
+
+  async proposeInstall(
+    proposal: InstallProposalRequest
```

Comment:
Is this pattern of making an interface for function arguments really necessary?

Reply:
Not normally, but in this case i think it's better to use an interface as opposed to 5+ positional arguments.